mythril/src/acpi/madt.rs

```diff
@@ -251,6 +253,44 @@ impl Ics {
         }
     }
 
+    /// Encode into the byte sequence
+    pub fn encode(&self) -> Result<Vec<u8>> {
```

Comment:
I'd really rather we structure this stuff so that we are given a buffer and write into it. Ultimately I'd like to move away from mythril requiring Alloc, so we should avoid using Vec wherever possible

Reply:
Yeah I can work on that.